app/models/foreman_azure_rm/azure_rm.rb

```diff
@@ -104,230 +124,136 @@ def available_subnets
-          vhd_path:                        args[:image_id],
-          os_disk_caching:                 args[:os_disk_caching],
-          data_disks:                      args[:volumes_attributes],
-          os_disk_size:                    args[:os_disk_size],
-          premium_os_disk:                 args[:premium_os_disk],
+      vm             = create_managed_virtual_machine(
+        name:                            args[:azure_vm][:vm_name],
+        location:                        args[:azure_vm][:location],
+        resource_group:                  args[:azure_vm][:resource_group],
+        vm_size:                         args[:azure_vm][:vm_size],
+        username:                        args[:azure_vm][:username],
+        password:                        args[:azure_vm][:password],
+        ssh_key_data:                    args[:azure_vm][:ssh_key_data],
+        disable_password_authentication: disable_password_auth,
+        network_interface_card_ids:      nics.map(&:id),
+        platform:                        args[:azure_vm][:platform],
+        vhd_path:                        args[:image_id],
+        os_disk_caching:                 args[:azure_vm][:os_disk_caching],
+        premium_os_disk:                 args[:azure_vm][:premium_os_disk],
+        custom_data:                     args[:azure_vm][:user_data],
+        script_command:                  args[:azure_vm][:script_command],
+        script_uris:                     args[:azure_vm][:script_uris],
       )
-      vm_hash                  = Fog::Compute::AzureRM::Server.parse(vm)
-      vm_hash[:password]       = args[:password]
-      vm_hash[:platform]       = args[:platform]
-      vm_hash[:puppet_master]  = args[:puppet_master]
-      vm_hash[:script_command] = args[:script_command]
-      vm_hash[:script_uris]    = args[:script_uris]
-      client.create_vm_extension(vm_hash)
-      client.servers.new vm_hash
-        # fog-azure-rm raises all ARM errors as RuntimeError
-    rescue Fog::Errors::Error, RuntimeError => e
-      Foreman::Logging.exception('Unhandled Azure RM error', e)
-      destroy_vm vm.id if vm
-      raise e
+
+      create_vm_extension(args)
+      # vm object using azure_vm
+      return_vm = AzureRMCompute.new(azure_vm: vm, sdk: sdk)
     end
 
     def destroy_vm(uuid)
+      #vm.azure_vm because that's the azure object and vm is the wrapper
       vm           = find_vm_by_uuid(uuid)
-      raw_model    = client.get_virtual_machine(vm.resource_group, vm.name)
-      os_disk_name = raw_model.storage_profile.os_disk.name
-      data_disks   = raw_model.storage_profile.data_disks
+      vm_name      = vm.name
+      rg_name      = vm.azure_vm.resource_group
+      os_disk      = vm.azure_vm.storage_profile.os_disk
+      data_disks   = vm.azure_vm.storage_profile.data_disks
       nic_ids      = vm.network_interface_card_ids
-      # In ARM things must be deleted in order
+
+      if sdk.get_vm(rg_name, vm_name).present?
```

Comment:
When will `find_vm_by_uuid` return a vm, but `sdk.get_vm` will fail?

Reply:
It behaves this way when a vm is present in foreman's hosts entry but not really in azure portal.